improver/calibration/ensemble_calibration.py

```diff
@@ -1033,10 +1129,14 @@ def _calculate_location_parameter_from_realizations(self, optimised_coeffs):
         # ensemble realizations. The number of b and X terms depends upon the
         # number of ensemble realizations. In this case, b = beta^2.
         beta_values = np.array([], dtype=np.float32)
-        for key in optimised_coeffs.keys():
-            if key.startswith("beta"):
-                beta_values = np.append(beta_values, optimised_coeffs[key])
-        a_and_b = np.append(optimised_coeffs["alpha"], beta_values ** 2)
+        for coeff_cube in self.coefficients_cubelist:
+            if coeff_cube.name() == "emos_coefficient_beta":
+                for coeff_slice in coeff_cube.slices_over("realization"):
+                    beta_values = np.append(beta_values, coeff_slice.data)
```

Comment:
Is this not just equivalent to:
```
beta_values = self.coefficients_cubelist.extract_strict(
    "emos_coefficient_beta"
).data.copy()
```
As even for spatially varying beta values we would expect realization to be the leading coordinate?

Reply:
I've replaced the `for` loop with the code that you've suggested.